app/client/cli/servicer.go

```diff
@@ -119,24 +141,23 @@ func validateServicer(session *rpc.Session, servicerAddress string) (*rpc.Protoc
 	return nil, fmt.Errorf("Error getting servicer: address %s does not match any servicers in the session %d", servicerAddress, session.SessionNumber)
 }
 
-// ADDTEST: add either unit tests or integration tests using kvstore
-// getSessionFrom uses the client-side session cache to retrieve a session for app/chain combination at the provided height, if one has already been retrieved and cached.
-func getSessionFromCache(sessionCache *cache.SessionCache, appAddress, chain string, height int64) (*rpc.Session, error) {
-	if sessionCache == nil {
-		return nil, fmt.Errorf("got nil session cache")
+// getSessionFromCache uses the client-side session cache to fetch a session for app+chain combination at the provided height, if one has already been retrieved and cached.
+func getSessionFromCache(c *cache.SessionCache, appAddress, chain string, height int64) (*rpc.Session, error) {
```

Comment:
OPTIONAL NIT: No stron opinion here, but do you think `c` should be a parameter or should we use the local variable given that `sessionCache` is initialized in `init`?

I can easily argue both sides here but curious to hear what you think.

Reply:
The initialization of `sessionCache` in `init` has added a package-level variable, which I think we should eventually refactor to build a clean(er) module/submodule for the CLI, with commands simply using the initialized CLI. TODO items have been previously added for this [here](https://github.com/pokt-network/pocket/blob/93f801f14dce6bc082b82d6305731566ad47d839/app/client/cli/servicer.go#L284) and [here](https://github.com/pokt-network/pocket/blob/93f801f14dce6bc082b82d6305731566ad47d839/app/client/cli/servicer.go#L86-L90). It seems you are already leaning that way based on [the review comment above](https://github.com/pokt-network/pocket/pull/888#discussion_r1269949703)

In the meantime, asking the cache to be explicitly passed to `getSessionFromCache` is an attempt (though admittedly not a high-impact one) to minimize the scattering of dependency on package-level variables throughout the code.

Setting this to resolved, but please LMK if this needs to be revisited in this PR.